File: libs/acc_magnitude/acc_magnitude.ts
// Icon unicode characters can be found at: http://fontawesome.io/icons/
//% weight=100 color=#c2b711 icon="\uf1ec" block="Accelerometer extra" advanced=false
namespace accMagnitude {

    /**
     * Example TS function for extension.
     */
    //% blockId=accmagnitude_example
    //% block
    export function example(): number {
        return 10;
    }

    /**
     * Get the accelerometer magnitude.
     *
     * This function is a shim and the micro:bit runs the C++ version.
     */
    //% blockId=accmagnitude_get_magnitude
    //% block
    //% shim=accMagnitude::get_magnitude
    export function getMagnitude(): number {
        return 1234;
    }
}

File: libs/acc_magnitude/acc_magnitude.cpp
#include "pxt.h"

using namespace pxt;

namespace accMagnitude {
    //%
    int get_magnitude() {
        int x = uBit.accelerometer.getX();
        int z = uBit.accelerometer.getY();
        int y = uBit.accelerometer.getZ();
        return (int)sqrtf((x * x) + (y * y) + (z * z));
    }
}
